sky/backends/backend_utils.py

```diff
@@ -2783,3 +2783,34 @@ def check_rsync_installed() -> None:
                 ' it is not installed. For Debian/Ubuntu system, '
                 'install it with:\n'
                 '  $ sudo apt install rsync') from None
+
+
+def wait_and_terminate_csync(cluster_name: str) -> None:
+    """
+    Terminatese all the CSYNC process running in each node after
+    waiting for the sync process launched by CSYNC complete.
+
+    Args:
+        cluster_name: Cluster name (see `sky status`)
+    """
+    record = global_user_state.get_cluster_from_name(cluster_name)
+    if record is None:
+        return
+    handle = record['handle']
+    if not isinstance(handle, backends.CloudVmRayResourceHandle):
+        return
+    ip_list = handle.external_ips()
```

Comment:
For cluster that is in INIT mode, this `external_ips()` will try to fetch the IPs and may fail if the cluster is not actually launched (probably due to `sky launch` and ctrl-c), causing failure for terminating the cluster.

We should make this function go pass if the IP fetching fails.

Reply:
Also, this can significantly increase the termination time if the csync just started on a normal cluster that is going to be terminated. It should be better to have better logging for it.

Reply:
@Michaelvll I added error handling when `external_ips` fails to fetch IP when the cluster is in `INIT` mode. 

Regards to the logging, do you have a suggestion on how to do the logging when the progress bar is running? I tried with two ways:
1. default `logger.info` shows redundant progress bar due to the logging
```
$ sky stop sky-2d0b-gcpuser -y
Stopping 1 cluster ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   0% -:--:--I 08-24 05:53:57 backend_utils.py:2845] CSYNC termination initiated. If a sync process is currently running, we will wait for it to complete before terminating the CSYNC daemon...
I 08-24 05:53:57 backend_utils.py:2845] 
Stopping 1 cluster ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   0% -:--:--

```
2. Using `rich.logging.RichHandler` manges to separate the progress bar and the logging, but the logging style changes compared to our default logging:
```
$ sky stop sky-2d0b-gcpuser -y
08-24 05:47:01 INFO     08-24 05:47:01 sky.backends.backend_utils:2846] CSYNC termination initiated. If a sync process is currently running, we will wait for it to complete before           backend_utils.py:2846
                        terminating the CSYNC daemon...                                                                                                                                                            
                                                                                                                                                                                                                   
Stopping 1 cluster ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
```

Reply:
@Michaelvll What do you suggest on this issue?

Reply:
This will be resolved after completing #2504